### PR TITLE
Create sub package for cronet

### PR DESCRIPTION
**Pull request by veblush**
Since cronet is not in core component and won't be, any target for cronet cannot be in main package. Therefore main package is refactored not to have cronet targets and new cronet package is created.

- Create new cronet package including all cronet targets.
  (Remove all cronet targets from main package.)

- Change objective_c/Cronet from a file to a target.

- Remove cronet function from gRPC++ since it's not done.

### CODE REVIEW

File: BUILD [hunk 1]
```diff
@@ -323,39 +323,36 @@ grpc_cc_library(
 )
 
 grpc_cc_library(
-    name = "grpc",
+    name = "grpc_init",
```
Comment (muxi) [inline BUILD]: The `srcs` is duplicate with the next target `grpc`. Is this intentional? How about have a dependency from `grpc` to `grpc_init` and remove `init.cc` from `grpc`

File: src/core/ext/transport/cronet/BUILD [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    ],
+)
+
+grpc_cc_library(
+    name = "grpc_transport_cronet_client_secure",
+    srcs = [
+        "client/secure/cronet_channel_create.cc",
+        "transport/cronet_api_dummy.cc",
+        "transport/cronet_transport.cc",
+        "transport/cronet_transport.h",
```
Comment (muxi) [inline src/core/ext/transport/cronet/BUILD]: `.h` should be in `hdrs` section

File: src/core/ext/transport/cronet/plugin_registry/grpc_cronet_plugin_registry.cc [hunk 3]
```diff
@@ -30,12 +30,9 @@ void grpc_client_channel_init(void);
 void grpc_client_channel_shutdown(void);
 
 void grpc_register_built_in_plugins(void) {
-  grpc_register_plugin(grpc_http_filters_init,
-                       grpc_http_filters_shutdown);
-  grpc_register_plugin(grpc_chttp2_plugin_init,
-                       grpc_chttp2_plugin_shutdown);
+  grpc_register_plugin(grpc_http_filters_init, grpc_http_filters_shutdown);
```
Comment (muxi) [inline src/core/ext/transport/cronet/plugin_registry/grpc_cronet_plugin_registry.cc]: Why😄? Does the formatter enforce difference column width in the two directories?

File: templates/test/core/surface/public_headers_must_be_c89.c.template [hunk 4]
```diff
@@ -25,13 +25,15 @@
       if platform_identifier in hdr:
         return True
     return False
+  def is_cronet_header(hdr):
+    return "cronet" in hdr
```
Comment (muxi) [inline templates/test/core/surface/public_headers_must_be_c89.c.template]: Would it better be case insensitive?

File: third_party/objective_c/Cronet/BUILD [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+licenses(["notice"])  # Apache v2
+
+package(default_visibility = ["//:__subpackages__"])
+
+# C Header files used by GRPC C Transport.
+cc_library(
+    name = "cronet_c_for_grpc",
```
Comment (muxi) [inline third_party/objective_c/Cronet/BUILD]: Name it `Cronet` to match the internal version?

**Comment by muxi** [comment]
@rmstar - please take a look. This might be in conflict with your Cronet work.

File: third_party/objective_c/Cronet/BUILD (hunk 5 above)
Comment (veblush) [reply third_party/objective_c/Cronet/BUILD]: It's the same name of internal version.

File: templates/test/core/surface/public_headers_must_be_c89.c.template (hunk 4 above)
Comment (veblush) [reply templates/test/core/surface/public_headers_must_be_c89.c.template]: Since this is for the public C header files which should be in lower case, I think case making it case insensitive is not necessary.

File: src/core/ext/transport/cronet/plugin_registry/grpc_cronet_plugin_registry.cc (hunk 3 above)
Comment (veblush) [reply src/core/ext/transport/cronet/plugin_registry/grpc_cronet_plugin_registry.cc]: This is the result of clang-formatter. Previous source was generated from the build tool and it doesn't go though the formatter.

File: third_party/objective_c/Cronet/BUILD (hunk 5 above)
Comment (muxi) [reply third_party/objective_c/Cronet/BUILD]: Oh I see.

File: src/core/ext/transport/cronet/BUILD (hunk 2 above)
Comment (veblush) [reply src/core/ext/transport/cronet/BUILD]: It sounds counter-intuitive but srcs can include header files which are supposed to be private. gRPC seems not to distinguish private and public headers so far, though.

File: BUILD (hunk 1 above)
Comment (veblush) [reply BUILD]: Because init.cc is referenced by many rules, it's a bit messy to clean them up. So let's remove grpc_init and grpc_cronet which haven't been used simply.

**Comment by veblush** [comment]
Muxi, thank you for the review!

File: src/cpp/client/cronet_credentials.cc [hunk 6]
```diff
@@ -1,64 +0,0 @@
...
-#include <grpcpp/security/credentials.h>
-
-#include <grpc/grpc_cronet.h>
-#include <grpcpp/channel.h>
-#include <grpcpp/support/channel_arguments.h>
-#include "src/cpp/client/create_channel_internal.h"
-
-namespace grpc {
-
-class CronetChannelCredentialsImpl final : public ChannelCredentials {
```
Comment (rmstar) [inline src/cpp/client/cronet_credentials.cc]: Why was this removed?

Comment (veblush) [reply src/cpp/client/cronet_credentials.cc]: It's removed because it belongs to the main BUILD file which cannot have cronet-related rules. It can have a separate BUILD file like Objective-C does. Even though it's removed in this CL, it's possible to revive it with the separate BUILD file.

Comment (rmstar) [reply src/cpp/client/cronet_credentials.cc]: Can you please add it back somewhere? I have some work in progress (Cronet C++ tests) that depends on the existence of this code.

Comment (rmstar) [reply src/cpp/client/cronet_credentials.cc]: Also need to ensure that cronet_credentials.cc is included in gRPC-C++.podspec as iOS tests will be using cocoapods to build Cronet C++

Comment (veblush) [reply src/cpp/client/cronet_credentials.cc]: This file is back again but not included in BUILD so please make your own rule in the separate BUILD file.

Comment (rmstar) [reply src/cpp/client/cronet_credentials.cc]: Thanks! Please add a header that defines `CronetChannelCredentials` as it has been removed from credentials.h. Also, please verify that cronet_credentials.cc and the new header that includes `CronetChannelCredentials` are available in `gRPC-C++.podspec`.

Comment (veblush) [reply src/cpp/client/cronet_credentials.cc]: Got it back again.

Comment (veblush) [reply src/cpp/client/cronet_credentials.cc]: New header file for cronet is added but it'd be better to integrate with podspec later since it should be refactored somehow not to be included in the main BUILD file.

File: src/core/ext/transport/cronet/BUILD [hunk 7]
```diff
@@ -0,0 +1,65 @@
+# gRPC Bazel BUILD file.
+#
+# Copyright 2016 gRPC authors.
```
Comment (rmstar) [inline src/core/ext/transport/cronet/BUILD]: 2019

File: src/core/ext/transport/cronet/client/secure/cronet_channel_create.h [hunk 8]
```diff
@@ -0,0 +1,39 @@
+/*
+ *
+ * Copyright 2016 gRPC authors.
```
Comment (rmstar) [inline src/core/ext/transport/cronet/client/secure/cronet_channel_create.h]: 2019

File: src/cpp/client/cronet_credentials.cc (hunk 6 above)
Comment (veblush) [reply src/cpp/client/cronet_credentials.cc]: gRPC-C++.podspec now has cronet files.
